Libraries/LibC/netdb.cpp

```diff
@@ -219,13 +276,77 @@ struct servent* getservbyport(int port, const char* protocol)
 }
 void setservent(int stay_open)
 {
-    dbg() << "FIXME: getservbyname(\"" << stay_open << "\")";
-    ASSERT_NOT_REACHED();
+    if(!services_file) {
+        services_file = fopen(serv_file_loc, "r");
+
+        if(!services_file) {
+            perror("error opening services file");
+            return;
+        }
+    }
+    rewind(services_file);
+    keep_service_file_open = stay_open;
 }
 void endservent()
 {
-    dbg() << "FIXME: getservbyname()";
-    ASSERT_NOT_REACHED();
+    if(!services_file) {
+        return;
+    }
+    fclose(services_file);
+    services_file = nullptr;
+    keep_service_file_open = 0;
+}
+
+//Fill the service entry buffer with the information contained in the currently read line, returns true if successfull, false if failure occurs.
+static bool fill_getserv_buffer(char* line, ssize_t read)
+{
+    //Splitting the line by tab delimiter and filling the servent buffers name, port, and protocol members.
+    auto split_line = String(line, read).split('\t');
+    strncpy(__getserv_name_buffer, split_line[0].characters(), split_line[0].length());
+    __getserv_buffer.s_name = __getserv_name_buffer;
+    
+    auto port_protocol_split = String(split_line[1]).split('/');
+    bool conversion_checker;
+    __getserv_buffer.s_port = port_protocol_split[0].to_int(conversion_checker);
+    
+    if(!conversion_checker) {
+        return false;
+    }
+    strncpy(__getserv_protocol_buffer, port_protocol_split[1].characters(), port_protocol_split[1].length());
+    __getserv_buffer.s_proto = __getserv_protocol_buffer;
+
+    //If there are aliases for the service, we will fill the alias list buffer with allocated strings for the alias names.
+    if(split_line.size() > 2 && !split_line[2].starts_with('#')) {
+        
+        auto split_aliases = String(split_line[2]).split(' ');
+
+        //The servent alias list has been utilized, we must free the previously allocated memory.
+        if(servent_alias_flag) {
+            
+            char** allocated_location = __getserv_alias_list_buffer;
+            while(*allocated_location) {
+                free(*allocated_location);
+                *allocated_location = 0;
+                allocated_location++;
+            }
+        }
+        for(size_t i = 0; i < split_aliases.size(); i++) {
+                __getserv_alias_list_buffer[i] = strdup(split_aliases[i].characters());
```

Comment:
Can we just use a C++ data structure with `String` and `Vector` as backing for `servent` (and synthesize a `servent` on demand when returning one, with pointers into our C++ structures)?
This seems overly complicated :)